ctrd/image.go

```diff
@@ -251,21 +251,24 @@ func (c *Client) PushImage(ctx context.Context, ref string, authConfig *types.Au
 	return nil
 }
 
-// FetchImage fetches image content from the remote repository.
-func (c *Client) FetchImage(ctx context.Context, name string, refs []string, authConfig *types.AuthConfig, stream *jsonstream.JSONStream) (containerd.Image, error) {
-	wrapperCli, err := c.Get(ctx)
+// ResolveImage attempts to resolve the image reference into a available reference and resolver.
+func (c *Client) ResolveImage(ctx context.Context, nameRef string, refs []string, authConfig *types.AuthConfig, opts docker.ResolverOptions) (remotes.Resolver, string, error) {
+	resolver, availableRef, err := c.getResolver(ctx, authConfig, nameRef, refs, opts)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get a containerd grpc client: %v", err)
+		return nil, "", err
```

Comment:
I think we should check the status and make sure that it is 404

Reply:
when resolveImage return err, means that it can't find available ref to use.